### DISCUSSION TITLE
The Potential for SSH Integration in Isomorphic-Git (or alternatives) and Mobile Environments for Obsidian-git?

### DISCUSSION THREAD

**Discussion by chrisoutwright** [Vinzent03/obsidian-git#526] [General]
I understand that the issue is that the isomorphic-git library is a pure JavaScript implementation of Git, specifically designed to run in browser and Node.js environments.  It is its current lack of support for SSH authentication. This limitation stems from technical complexities and security issues associated with implementing SSH within a browser or JavaScript-based environment.

Is this the main limiting factor on why ssh protocol cannot be supported on android?:

> to authenticate the remote computer and allow the remote computer to authenticate the user, if necessary. 
> One of the main reasons is the secure management of private keys, which are essential for SSH authentication. 
> They must be stored securely and loaded into memory only when necessary. This presents a considerable challenge in a web-based context, where storage and memory management mechanisms aren't as robust or secure as in native applications.

Or is it because?:

> SSH incorporates complex network protocols that can be challenging to implement efficiently and securely in JavaScript, which is not as performant or low-level as languages traditionally used for this type of task, such as C or C++.

**Comment by Vinzent03** [comment]
I don't know the reason for lack of ssh support in isomorphic-git. You may open an issue in their repo.
